src/index.test.ts

```diff
@@ -1,6 +1,1431 @@
+      const message = {
+        from: {
+          name: 'Cow',
+          wallet: '0xCD2a3d9F938E13CD947Ec05AbC7FE734Df8DD826',
+        },
+        to: {
+          name: 'Bob',
+          wallet: '0xbBbBBBBbbBBBbbbBbbBbbbbBBbBbbbbBbBbbBBbB',
+        },
+        contents: null,
+      };
+
+      expect(
+        sigUtil.TypedDataUtils.encodeData(
+          primaryType,
+          message,
+          types,
+          'V3',
+        ).toString('hex'),
+      ).toMatchSnapshot();
+    });
+
+    it('should encode data with a dynamic property set to undefined', function () {
+      const types = {
+        Person: [
+          { name: 'name', type: 'string' },
+          { name: 'wallet', type: 'address' },
+        ],
+        Mail: [
+          { name: 'from', type: 'Person' },
+          { name: 'to', type: 'Person' },
+          { name: 'contents', type: 'string' },
+        ],
+      };
+      const primaryType = 'Mail';
+      const message = {
+        from: {
+          name: 'Cow',
+          wallet: '0xCD2a3d9F938E13CD947Ec05AbC7FE734Df8DD826',
+        },
+        to: {
+          name: 'Bob',
+          wallet: '0xbBbBBBBbbBBBbbbBbbBbbbbBBbBbbbbBbBbbBBbB',
+        },
+        contents: undefined,
+      };
+
+      expect(
+        sigUtil.TypedDataUtils.encodeData(
+          primaryType,
+          message,
+          types,
+          'V3',
+        ).toString('hex'),
+      ).toMatchSnapshot();
+    });
+
+    it('should throw an error when a custom type property is set to null', function () {
+      const types = {
+        Person: [
+          { name: 'name', type: 'string' },
+          { name: 'wallet', type: 'address' },
+        ],
+        Mail: [
+          { name: 'from', type: 'Person' },
+          { name: 'to', type: 'Person' },
+          { name: 'contents', type: 'string' },
+        ],
+      };
+      const primaryType = 'Mail';
+      const message = {
+        to: null,
+        from: {
+          name: 'Cow',
+          wallet: '0xCD2a3d9F938E13CD947Ec05AbC7FE734Df8DD826',
+        },
+        contents: 'Hello, Bob!',
+      };
+
+      expect(() =>
+        sigUtil.TypedDataUtils.encodeData(
+          primaryType,
+          message,
+          types,
+          'V3',
+        ).toString('hex'),
+      ).toThrow(`Cannot read property 'name' of null`);
+    });
+
+    it('should encode data with a custom type property set to undefined', function () {
+      const types = {
+        Person: [
+          { name: 'name', type: 'string' },
+          { name: 'wallet', type: 'address' },
+        ],
+        Mail: [
+          { name: 'from', type: 'Person' },
+          { name: 'to', type: 'Person' },
+          { name: 'contents', type: 'string' },
+        ],
+      };
+      const primaryType = 'Mail';
+      const message = {
+        from: {
+          name: 'Cow',
+          wallet: '0xCD2a3d9F938E13CD947Ec05AbC7FE734Df8DD826',
+        },
+        contents: 'Hello, Bob!',
+      };
```

Comment:
This test case calls from a custom type value with a property "set to undefined", whereas the property in question (`to`) is actually not there at all. The test would be slightly more readable if the message object included `to: undefined`. It would also match what you've done in other test cases.

Reply:
The same goes for the corresponding test case in the V4 suite.

Reply:
I think they're equivalent functionally, but yes that might be easier to read. That sounds good to me.

Reply:
Done in https://github.com/MetaMask/eth-sig-util/pull/164/commits/6e313ad055739f8c536763b0fecc884b33f0af7b